syntax = "proto3";

package game;

service Connect4Service {
  rpc NewGame(Connect4.NewGameReq) returns (Connect4.Game) {}
  rpc GetMoves(Connect4.Game) returns (Connect4.MoveList) {}
  rpc MakeMove(Connect4.MakeMoveReq) returns (Connect4.Empty) {}
}

message Connect4 {
  message Empty {}

  message Game {
    optional uint32 id = 1;
  }

  message Move {
    optional uint32 row = 1;
    optional uint32 col = 2;
  }

  message MoveList {
    repeated Move moves = 1;
  }

  message NewGameReq {
    optional uint32 serverPlayer = 1;
    optional uint32 difficulty = 2;
  }

  message MakeMoveReq {
    optional Game game = 1;
    optional Move move = 2;
  }
}
